policy-report/image-vuln-adapter/pkg/cmd/write.go

```diff
@@ -0,0 +1,84 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	policyreport "github.com/kubernetes-sigs/wg-policy-prototypes/policy-report/image-vuln-adapter/pkg/apis/wgpolicyk8s.io/v1alpha2"
+
+	client "github.com/kubernetes-sigs/wg-policy-prototypes/policy-report/image-vuln-adapter/pkg/generated/v1alpha2/clientset/versioned"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog"
+)
+
+func Write(r []policyreport.PolicyReport, kubeconfigPath string) error {
+
+	var kubeconfig *rest.Config
+
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		cfg, err = clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+		if err != nil {
+			klog.Fatalf("Error building kubeconfig: %s", err.Error())
+			fmt.Println("error: ", err)
+			return nil
+		}
+	}
+	kubeconfig = cfg
+	clientset, err := client.NewForConfig(kubeconfig)
+	if err != nil {
+		fmt.Println("error: ", err)
+			return nil
+	}
+	
+	for _, report := range r {
+		var r *policyreport.PolicyReport
+		r = &report
+		policyReport := clientset.Wgpolicyk8sV1alpha2().PolicyReports(r.Namespace)
+		// Check for existing Policy Reports
+		result, getErr := policyReport.Get(context.Background(), r.Name, metav1.GetOptions{})
+		fmt.Println(result)
+		// Create new Policy Report if not found
+		if errors.IsNotFound(getErr) {
+			fmt.Println("creating cluster-wide policy report...")
```

Comment:
Is this a namespaced or a cluster-wide policy report?

Reply:
That part was copied, it's the creating for the namespace